test failure when exact equality expected for floating point values

## Expected behavior ##

After building mdanalysis with python3 (both 3.9 and 3.10), tests fails, specifically `test_transform_StoR_pass` in testsuite/MDAnalysisTests/lib/test_distances.py.

The problem is that `assert_equal` is being used to test equality, which is not really appropriate for testing the equality of floating point numbers.  [`assert_allclose`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose) should be used instead. (documentation for assert_almost_equal recommends using assert_allclose instead).

Tests pass with assert_allclose with no precision specified (i.e. default rtol=1e-07 is satisfactory, at least on amd64).

## Actual behavior ##

The error message is
```
=================================== FAILURES ===================================
______________________ test_transform_StoR_pass[float32] _______________________

coord_dtype = <class 'numpy.float32'>

    @pytest.mark.parametrize('coord_dtype', (np.float32, np.float64))
    def test_transform_StoR_pass(coord_dtype):
        box = np.array([10, 7, 3, 45, 60, 90], dtype=np.float32)
        s = np.array([[0.5, -0.1, 0.5]], dtype=coord_dtype)
    
        original_r = np.array([[ 5.75,  0.36066014, 0.75]], dtype=np.float32)
    
        test_r = distances.transform_StoR(s, box)
    
>       assert_equal(original_r, test_r)
E       AssertionError: 
E       Arrays are not equal
E       
E       Mismatched elements: 1 / 3 (33.3%)
E       Max absolute difference: 2.9802322e-08
E       Max relative difference: 8.263271e-08
E        x: array([[5.75   , 0.36066, 0.75   ]], dtype=float32)
E        y: array([[5.75   , 0.36066, 0.75   ]], dtype=float32)

/projects/mdanalysis/testsuite/MDAnalysisTests/lib/test_distances.py:73: AssertionError
______________________ test_transform_StoR_pass[float64] _______________________

coord_dtype = <class 'numpy.float64'>

    @pytest.mark.parametrize('coord_dtype', (np.float32, np.float64))
    def test_transform_StoR_pass(coord_dtype):
        box = np.array([10, 7, 3, 45, 60, 90], dtype=np.float32)
        s = np.array([[0.5, -0.1, 0.5]], dtype=coord_dtype)
    
        original_r = np.array([[ 5.75,  0.36066014, 0.75]], dtype=np.float32)
    
        test_r = distances.transform_StoR(s, box)
    
>       assert_equal(original_r, test_r)
E       AssertionError: 
E       Arrays are not equal
E       
E       Mismatched elements: 1 / 3 (33.3%)
E       Max absolute difference: 2.9802322e-08
E       Max relative difference: 8.263271e-08
E        x: array([[5.75   , 0.36066, 0.75   ]], dtype=float32)
E        y: array([[5.75   , 0.36066, 0.75   ]], dtype=float32)

/projects/mdanalysis/testsuite/MDAnalysisTests/lib/test_distances.py:73: AssertionError
```

## Code to reproduce the behavior ##

The triggering code is

[    assert_equal(original_r, test_r)
](https://github.com/MDAnalysis/mdanalysis/blob/db0c3d3b28ffb1861b130d388b8a33f81c178239/testsuite/MDAnalysisTests/lib/test_distances.py#L74)

The test can be launched manually from source with
```
python3 -mpytest testsuite -k "test_transform_StoR_pass"
```

## Current version of MDAnalysis ##

- Which version are you using?  2.0.0
- Which version of Python?  Python 3.9.10  (same error with 3.10.2)
- Which operating system?  Linux (Debian unstable)

numpy 1.21.5
scipy 1.7.3

--- a/testsuite/MDAnalysisTests/lib/test_distances.py
+++ b/testsuite/MDAnalysisTests/lib/test_distances.py
@@ -71,7 +71,7 @@ def test_transform_StoR_pass(coord_dtype):
 
     test_r = distances.transform_StoR(s, box)
 
-    assert_equal(original_r, test_r)
+    assert_allclose(original_r, test_r)
 
 
 def test_capped_distance_noresults():

